### PR TITLE
Some basic docs to get started with

### PR DESCRIPTION
**Pull request by wibblymat**
R: @kaycebasques @gauntface
CC: @petele

### CODE REVIEW

**Comment by gauntface** [comment]
Would it be worth changing this to just be steps to use rather than an explainer of the API?

I'd expect API docs to explain the API itself and the README be the rough steps to get this working.
1. install the library
   
   ```
   npm install ....
   ```
2. Send a payload to a subscription.
   
   ```
   // Subscription should be stored on your backend {endpoint: '.....', keys: {auth: '....', p256dh: '...'}}
   sendWebPush(data, subscription);
   ```

Question regarding API - was there a reason for not handling the authentication header inside the library rather than the developer having to manage it?

**Comment by wibblymat** [comment]
`addAuthToken` may be a bad name, then... it is the method you use to tell the library what your API key is for that service. You call it once in your application, not per message sent. So for GCM I'd do:

```
webpush.addAuthToken('.googleapis.com/gcm', '<string I copied from console.developers.google.com>');

subscriptions.forEach(sub => webpush.sendWebPush('whatever', subscription));
```

Does that make sense? The actual `Authentication: key=BLAH` header is added to the POST request by the library.

File: README.md [hunk 1]
```diff
@@ -5,6 +5,82 @@ This library provides the functions necessary to encrypt a payload for sending
...
+Overview
+--------
+
+Install the module using npm:
+
+`npm install web-push-encryption`
+
+Require the module:
+
+`const webpush = require('web-push-encryption');`
```
Comment (petele) [inline README.md]: Nit: I thought the plan for ES2015 usage was to use it in the libraries, but keep samples and docs in plain JS?

File: README.md [hunk 2]
```diff
@@ -5,6 +5,82 @@ This library provides the functions necessary to encrypt a payload for sending
...
+
+Send a message:
```
Comment (petele) [inline README.md:36]: Seriously, that's all I need to do, I don't need to create any keys, certificates, sign anything, jump on one leg three times while patting my head? If so, I <3 this.

File: README.md [hunk 3]
```diff
@@ -5,6 +5,82 @@ This library provides the functions necessary to encrypt a payload for sending
...
+requests. You can specify which tokens to send for which push providers. Both
+`pattern` and `token` are Strings. When sending the message to the endpoint, the
+endpoint is matched against each pattern that has been set. If any pattern is a
+substring of the endpoint then the associated token will be sent in the request.
+
+For example, to set the token for GCM you could use a pattern of
+`https://android.googleapis.com/gcm`.
+
+**encrypt**
+`.encrypt(message, subscription)`
```
Comment (petele) [inline README.md]: doesn't `sendWebPush` do this? why do I need to have this?

**Comment by petele** [comment]
Added a few comments, if you gave this to me today and said go, it would take me too long to figure out how to use it. I want a quick start explanation to get me going immediately, then go from there.

File: README.md (hunk 1 above)
Comment (gauntface) [inline README.md]: never heard this being discussed but it's a good point - it probably should be.

**Comment by gauntface** [comment]
Ta for the explanation, still not entirely convinced by this API though.

1.) If it's only used by GCM, then should it just be of addGCMAuthToken('<string I copied from console.developers.google.com>');
2.) If it is for other push providers or will be in the future - it presumably would just be a server key or something to that affect and would be shared across all push providers meaning you could have

webpush.addGCMAuthToken('<string I copied from console.developers.google.com>');
webpush.addAuthToken('<shared server auth token>');

Essentially - I just really dislike the pattern requirement - feels to error prone and boilerplate.

SIDE NOTE: This should probably be moved to an issue and doesn't block these this docs PR.

**Comment by gauntface** [comment]
+1 Peters comments.

I think keep this on the bare minimum titled getting started to just install and use the library for Chrome  and Firefox.

We can either than add further comments in API docs OR seperate sections. For now I'd go minimal and start asking people for feedback and iterate on it.

**Comment by gauntface** [comment]
Moved auth token comments to: https://github.com/GoogleChrome/push-encryption-node/issues/6 would appreciate any feedback.

**Comment by wibblymat** [comment]
@gauntface @petele I simplified the docs, removing the API reference.

I also simplified the auth token business - auth token is now a parameter of sendWebPush and we push the process onto the developer. This came up in the code review for the Go version, too, and in the end it seemed like it was actually easier for developers to grok what was going on if they were left in control.

File: README.md [hunk 4]
```diff
@@ -9,6 +9,43 @@ This library provides the functions necessary to encrypt a payload for sending
...
+server must encrypt the payload or the receiving browser will reject it.
+
+This library implements the necessary encryption as a Node module.
+
+Overview
+--------
+
+Install the module using npm:
+
+`npm install web-push-encryption`
```
Comment (gauntface) [inline README.md:30]: Is this the name we are actually published on? Does it make sense to make it slightly more generic if we can support encrypted push and just tickles?

**Comment by gauntface** [comment]
LGTM

Commented on the naming of the module, but that can be moved to an issue.

File: README.md (hunk 4 above)
Comment (wibblymat) [inline README.md:30]: I wanted `web-push` but Mozilla got there first :)

Will open an issue.

**Pull request merged**